### PR TITLE
Signup merged down parts 1-5

### PR DESCRIPTION
**Pull request by chrisnojima**
Now merged down. Waiting for a second thumb

---------------------

- [x] Changed several engine methods from taking a bunch of params to an object so its more clear what's what and avoids passing a bunch of `null,null,null` type stuff
- [x] Promise rpcs and engine sagas take a waiting key instead of a waiting callback or waiting action creator
- [x] Added a changeWaiting action so you can just pass a boolean if you want an increment or not

This is part of a 4 part pr. It won't work until the 4th part: 
1. https://github.com/keybase/client/pull/12524
1. https://github.com/keybase/client/pull/12525
1. https://github.com/keybase/client/pull/12526
1. https://github.com/keybase/client/pull/12527

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

**Comment by chrisnojima** [comment]
- [ ] merge other 3 first

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is the merged down parts 1-5 of the signup pr. still looking for a second thumb (first was @akalin s)

File: shared/actions/signup.js [hunk 1]
```diff
@@ -1,366 +1,198 @@
...
-    })
-    return p
-  }
+        },
+        waitingKey: Constants.waitingKey,
+      })
+      yield Saga.put(SignupGen.createSignedup())
+    } catch (error) {
+      yield Saga.put(
+        SignupGen.createSignedupError({error: new HiddenString(`Cant signup, try again? ${error.desc}`)})
```
Comment (buoyad) [inline shared/actions/signup.js]: The page already renders `Ah Shoot! something went wrong...`, so the `Cant signup, try again?` is redundant here.
image

File: shared/actions/signup.js [hunk 2]
```diff
@@ -1,366 +1,198 @@
...
-              // Do not add a gpg key for now
-              response.result(false)
-            },
-            'keybase.1.loginUi.displayPrimaryPaperKey': ({sessionID, phrase}, response) => {
-              paperKeyResponse = response
-              dispatch(SignupGen.createShowPaperKey({paperkey: new HiddenString(phrase)}))
-              onDisplayPaperKey && onDisplayPaperKey()
-              dispatch(nextPhase())
-            },
+  return Saga.call(function*() {
```
Comment (buoyad) [inline shared/actions/signup.js]: We usually encode this pattern w/ `safeTakeEveryPure(..., actionCreatorsWithError)`, could you do that here?

**Pull request merged**